jormungandr/src/fragment/pool.rs

```diff
@@ -85,11 +59,18 @@ impl Pool {
                         .zip(fragments_exist_in_logs)
                         .filter(|(_, exists_in_logs)| !exists_in_logs)
                         .map(|(fragment, _)| fragment);
-                    let new_fragment_ids = pool.insert_all(new_fragments);
-                    let count = new_fragment_ids.len();
-                    let fragment_logs = new_fragment_ids
-                        .into_iter()
-                        .map(move |id| FragmentLog::new(id.into(), origin));
+                    let new_fragments = pool.insert_all(new_fragments);
+                    let count = new_fragments.len();
+                    let fragment_logs = new_fragments
+                        .iter()
+                        .map(move |fragment| FragmentLog::new(fragment.id().into(), origin))
+                        .collect::<Vec<_>>();
+                    for fragment in new_fragments {
+                        let network_msg = NetworkMsg::Propagate(PropagateMsg::Fragment(fragment));
+                        network_msg_box.try_send(network_msg).unwrap_or_else(|err| {
+                            error!(logger, "cannot propagate fragment to network: {}", err)
+                        });
```

Comment:
`MessageBox` wraps `tokio::sync::mpsc::Sender`, so using `try_send` this way can get messages dropped on the floor if the capacity of the buffer overflows.

Try to return `new_fragments` as the result of the poll future and add an `add_then` to process them in a sequence of futures returned by `network_msg_box.send(...)`.